client/src/includes/breadcrumbs.test.js

```diff
@@ -0,0 +1,70 @@
+import initCollapsibleBreadcrumbs from './breadcrumbs';
+
+describe('initCollapsibleBreadcrumbs', () => {
+  it('should do nothing if there is no breadcrumbs container', () => {
+    jest.spyOn(document, 'addEventListener').mockImplementation(() => {});
+    // Set up our document body
+    document.body.innerHTML =
+      '<div>' +
+      '  <span id="username" />' +
+      '  <button id="button" />' +
+      '</div>';
+    initCollapsibleBreadcrumbs();
+    // no event listeners registered
+    expect(document.addEventListener).not.toHaveBeenCalled();
+  });
+
+  describe('there is a breadcrumbs container present', () => {
+    jest.spyOn(document, 'addEventListener').mockImplementation(() => {});
+    it('should expand the breadcrumbs when clicked', () => {
+      // Set up our document body
+      document.body.innerHTML =
+        '<div data-slim-header>' +
+        '<div data-breadcrumb-next>' +
+        '  <button id="button" data-toggle-breadcrumbs >' +
+        '<svg aria-hidden="true">' +
+        '  <use id="use" href="#icon-breadcrumb-expand" />' +
+        ' </svg>' +
+        '</button>' +
+        '<nav aria-label="Breadcrumb">' +
+        '  <span id="username" data-breadcrumb-item />' +
+        '</nav>' +
+        '</div>' +
+        '</div>';
+      initCollapsibleBreadcrumbs();
+      // event listeners registered
+      expect(document.addEventListener).toHaveBeenCalled();
+      // click to expand the breadcumbs
+      document.getElementById('button').click();
+      // click to change the button icon
+      document.getElementById('button').click();
+      expect(
+        document.getElementById('button').getAttribute('aria-expanded'),
+      ).toBe('true');
+      expect(document.getElementById('use').getAttribute('href')).toBe(
+        '#icon-cross',
+      );
+    });
+
+    it('should collapse the breadcrumbs when clicked, if expanded', () => {
+      initCollapsibleBreadcrumbs();
```

Comment:
no need to call this again, as it is already initialised against your DOM.

Reply:
The reason for calling it was to initialize the DOM to its original state and then in the same test I expanded it first then collapsed it.
After removing it the DOM stays in the previous test state i.e in expanded and I only have to collapse it which does make the code shorter and more efficient.